### ISSUE TITLE
Selecting a destination from Favorites is non-responsive

### ISSUE THREAD

**Issue by sonora** [osmandapp/Osmand#8087]
I have several hundred Favorites configured. Selecting one via Menu button / MyPlaces / Favorites / Directions works fine.

But when, from the map screen, I tap the Directions button, then on the route dialog screen tap Set destination, then Favorites, I see a list of my Favorites sorted by distance. Trying to tap any one of them  is rather non-responsive. I have to tap 5-7 times before one is accepted and the route starts calculating. Looks like that Favorite list screen is real busy and has no time to respond to any taps. Maybe it is busy creating a complete list of all x-hundred entries, rather than responding to taps?

**Comment by sonora** [comment]
@max-klaus Looking at this fix, I think it may reference the wrong issue? Testing with this commit, I see no difference for this issue

**Comment by max-klaus** [comment]
So it may depend on number of favorites. Let me reopen it.

**Comment by sonora** [comment]
On second thought: Maybe you really fixed it :) , or it needs more research. Let me close it for now.

What I have now tried with the latest nightly: I have two categories of Favorites marked visible via Show on map: Cat A has about 150 entries, cat B 500. I disbled Show on map for cat B, and the issue disappeared. Then I re-enabled it, and the issue did not reappear.

Could have to do with recent GPX database version updates we had played with in recent nightlies.
